cluster.py

```diff
@@ -8,14 +10,33 @@
 logger = logging.getLogger(__name__)
 
 
+def get_cluster_lock(timeout: int = 10 * 60) -> socket.socket:
```

Comment:
I don't think it is good idea to use `lock` wording for the retry-mechanism.

Reply:
it's not retry mechanism, it is acquiring port - as long it is acquired it is locked for other processes